replace old chapter by new chapter in Q2-w1

mzijlema opened the pull request:
Added new chapter _Numerical methods for PDEs_ which replaces the old chapter entitled _PDEs and the Finite Volume Method_ in second quarter, first week.

mzijlema committed:
Update intro.md

Small changes to begin with.

mzijlema committed:
Create intro.md

Introduction to numerical methods for partial differential equations; these are the finite difference method (FDM) and the finite volume method.

mzijlema committed:
Update intro.md

extend introduction (v1)

mzijlema committed:
Update intro.md

Further extension of Introduction (v2)

mzijlema committed:
Create overview

Introduction to the numerical solution to PDEs.

mzijlema committed:
Update intro.md

Small modifications.

mzijlema committed:
Update overview

Small changes.

mzijlema committed:
Rename intro.md to intro_to_pdes.md

rename file intro.md into intr_to_pdes

mzijlema committed:
Update _toc.yml

First update Table of Contents with respect to Chapter Numerical Methods for PDEs

mzijlema committed:
Rename overview to overview.md

add extension md to this file

mzijlema committed:
Update intro_to_pdes.md

Major improvements of the text and equations

mzijlema committed:
Update overview.md

Modify the text. This is still incomplete.

mzijlema committed:
Update intro_to_pdes.md

small correction on second exercise

mzijlema committed:
Update intro_to_pdes.md

some modifications and correct references

mzijlema committed:
Update intro_to_pdes.md

small typos

mzijlema committed:
Update references.bib

add reference of book Andrews on PDEs

mzijlema committed:
Update intro_to_pdes.md

add reference of Andrews on PDEs
add important note on LOs

mzijlema committed:
Update credits.md

add credit on chapter Numerical Methods for PDEs

mzijlema committed:
Update overview.md

small modification of title
add credit

mzijlema committed:
Update credits.md

small update to credit Numerical Methods for PDEs

mzijlema committed:
Update intro_to_pdes.md

small updates

mzijlema committed:
Update credits.md

correct symbol ampersand

mzijlema committed:
Update credits.md

again a try to correct the ampersand symbol

mzijlema committed:
Update credits.md

ampersand problem

mzijlema committed:
Update credits.md

Okay, no ampersand. Instead another html link.

mzijlema committed:
Update intro_to_pdes.md

List of PDEs visually improved

mzijlema committed:
Update intro_to_pdes.md

Again improvement list of PDEs

mzijlema committed:
Update overview.md

small extension of introduction

mzijlema committed:
Update intro_to_pdes.md

small updates

mzijlema committed:
Update overview.md

mzijlema committed:
Update credits.md

small typo

mzijlema committed:
Update overview.md

Extension of this overview.

mzijlema committed:
Update overview.md

typos corrections

mzijlema committed:
Update overview.md

small changes

mzijlema committed:
Update overview.md

small typo

mzijlema committed:
Update intro_to_pdes.md

small changes

mzijlema committed:
Update intro_to_pdes.md

small updates

mzijlema committed:
Update intro_to_pdes.md

a very small change

mzijlema committed:
Create diffusion_equation

add section on discretization of the diffusion equation

mzijlema committed:
Update _toc.yml

add section diffusion equation

mzijlema committed:
Rename diffusion_equation to diffusion_equation.md

add md extension

mzijlema committed:
Update diffusion_equation.md

add title of section

mzijlema committed:
Update _toc.yml

small update

mzijlema committed:
Update diffusion_equation.md

extension of section Diffusion Equation

mzijlema committed:
Update diffusion_equation.md

further extensions

mzijlema committed:
Update diffusion_equation.md

further extensions

mzijlema committed:
Update diffusion_equation.md

further extensions

mzijlema committed:
Update diffusion_equation.md

further extensions

mzijlema committed:
Update diffusion_equation.md

small changes to correct compiling

mzijlema committed:
Update diffusion_equation.md

small corrections for referring equations

mzijlema committed:
Update diffusion_equation.md

small changes

mzijlema committed:
Update diffusion_equation.md

correction latex formula

mzijlema committed:
Update diffusion_equation.md

mzijlema committed:
Update diffusion_equation.md

small correction latex formula

mzijlema committed:
Update diffusion_equation.md

again small correction

mzijlema committed:
Update diffusion_equation.md

add solutions to the last two exercises

mzijlema committed:
Update diffusion_equation.md

further extensions

mzijlema committed:
Update diffusion_equation.md

small changes to improve readability

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update diffusion_equation.md

further extensions and small edits

mzijlema committed:
Update diffusion_equation.md

small typo

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update diffusion_equation.md

further extensions: matrix-vector notation

mzijlema committed:
Update diffusion_equation.md

small changes/edits

mzijlema committed:
Update diffusion_equation.md

add exercise on the solution of system of ODEs

mzijlema committed:
Update diffusion_equation.md

small edits on the exercise

mzijlema committed:
Update diffusion_equation.md

correct matrix

mzijlema committed:
Update diffusion_equation.md

small changes

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update diffusion_equation.md

further extension on time integration

mzijlema committed:
Update diffusion_equation.md

add edits and subsections

mzijlema committed:
Update diffusion_equation.md

major edits

mzijlema committed:
Update diffusion_equation.md

small changes

mzijlema committed:
Update diffusion_equation.md

further extension: consistency

mzijlema committed:
Update diffusion_equation.md

small typo

mzijlema committed:
Update diffusion_equation.md

small changes

mzijlema committed:
Update diffusion_equation.md

small changes

mzijlema committed:
Update diffusion_equation.md

small changes

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update diffusion_equation.md

small changes

mzijlema committed:
Update diffusion_equation.md

further extensions

mzijlema committed:
Update diffusion_equation.md

small changes

mzijlema committed:
Update diffusion_equation.md

small changes

mzijlema committed:
Update diffusion_equation.md

small changes

mzijlema committed:
Update diffusion_equation.md

further extension: add an exercise on truncation error on x=L and stability section

mzijlema committed:
Update diffusion_equation.md

further extension: stability based on positivity concept

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update overview.md

small edits

mzijlema committed:
Create advection_equation.md

add new paragraph: advection equation

mzijlema committed:
Update _toc.yml

add section Advection Equation to TOC

mzijlema committed:
Update advection_equation.md

small edits

mzijlema committed:
Update advection_equation.md

further extensions

mzijlema committed:
Update advection_equation.md

further extensions

mzijlema committed:
Update advection_equation.md

further extensions

mzijlema committed:
Update advection_equation.md

further extensions: upwind scheme

mzijlema committed:
Update advection_equation.md

dcases -> cases

mzijlema committed:
Update advection_equation.md

further extensions

mzijlema committed:
Update advection_equation.md

small edits

mzijlema committed:
Update advection_equation.md

some major edits

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update advection_equation.md

small edit

mzijlema committed:
Update advection_equation.md

further extensions: the leapfrog scheme

mzijlema committed:
Update advection_equation.md

small edits

mzijlema committed:
Update advection_equation.md

small edit

mzijlema committed:
Update advection_equation.md

small edits

mzijlema committed:
Update advection_equation.md

small edits

mzijlema committed:
Create advection_diffusion_eq.md

add section advection-diffusion equation

mzijlema committed:
Update _toc.yml

add section advection-diffusion equation to ToC

mzijlema committed:
Update intro_to_pdes.md

small edit: referring to section advection-diffusion equation

mzijlema committed:
Update advection_diffusion_eq.md

small edits

mzijlema committed:
Update advection_diffusion_eq.md

further extension: stationary advection-diffusion equation

mzijlema committed:
Update advection_diffusion_eq.md

small corrections

mzijlema committed:
Update advection_diffusion_eq.md

small changes

mzijlema committed:
Update advection_diffusion_eq.md

small edits

mzijlema committed:
Update advection_diffusion_eq.md

small edits

mzijlema committed:
Update advection_diffusion_eq.md

mzijlema committed:
Update advection_diffusion_eq.md

small edit

mzijlema committed:
Update advection_diffusion_eq.md

further extensions

mzijlema committed:
Update advection_diffusion_eq.md

further extensions

mzijlema committed:
Update advection_diffusion_eq.md

small edits

mzijlema committed:
Update advection_diffusion_eq.md

small edits

mzijlema committed:
Update advection_diffusion_eq.md

small edits

mzijlema committed:
Update advection_diffusion_eq.md

small edits

mzijlema committed:
Update advection_diffusion_eq.md

further extensions and small edits

mzijlema committed:
Update advection_diffusion_eq.md

further extension: instationary advection-diffusion equation

mzijlema committed:
Update advection_diffusion_eq.md

add solution to last question

mzijlema committed:
Update advection_diffusion_eq.md

typo

mzijlema committed:
Update advection_diffusion_eq.md

small edits

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update advection_equation.md

add LO's

mzijlema committed:
Update advection_equation.md

small edits

mzijlema committed:
Update overview.md

added references to different sections

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update advection_equation.md

small edits

mzijlema committed:
Update diffusion_equation.md

add comment on symmetric matrix

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update diffusion_equation.md

add comment on stability by means of spectral analysis

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update advection_equation.md

small edits

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update advection_diffusion_eq.md

small edits

mzijlema committed:
Update diffusion_equation.md

typo

mzijlema committed:
Update diffusion_equation.md

small edits

mzijlema committed:
Update advection_diffusion_eq.md

typo

mzijlema committed:
Update advection_diffusion_eq.md

typo

mzijlema committed:
Update advection_equation.md

small edit

mzijlema committed:
Update advection_equation.md

small edit

mzijlema committed:
Update advection_diffusion_eq.md

small edit

mzijlema committed:
Update advection_diffusion_eq.md

small edit

mzijlema committed:
Merge branch '2025-draft' into zijlema-review-q2-w1

fmeer committed:
Minor edits to PDE chapter

fmeer commented:
The chapter looks great. Very clear throughout. Sets a new standard for the book with learning objectives per page. Clean language and formatting. I made some small edits in https://github.com/TUDelft-MUDE/book/pull/153/commits/f0eb26ed61c5ef094678f9c12ff83821e8ab6213

A few points, if you agree @mzijlema and want me to process them, I could do that, but I did not want to do these without consulting you:
- [x] In other chapters we use bold notation for vectors and matrices. For consistency I would prefer to replace all instances of `\vec` with  `\mathbf`. Also: $\mathbf{A}$ and $\mathbf{I}$ for matrices
- [x] Reference to hydraulic engineering at the end of intro_to_pdes is too exclusive, we have a broader audience here
- [x] In the discussion on stability of the diffusion solution, it is stated that $T(x,t) \leq 0$. However, with general initiation conditions from (1.2) this is not necessarily the case. 
- [x] In the advection page, is it correct to replace "(especially in hydraulic engineering)" with "in engineering"? Otherwise maybe remove the statement. 
- [x] I would prefer to always write out "method of lines" instead of MOL for independent readability of the different Sections. 

Two bigger points for further potential improvement: 
- The chapter has quite a lot of content. An option could be to exclude Section 1.4 from study material for the exam and keep it in as material "for the interested reader". Then it is also accessible for HE students later on. Whether it is too much or not also depends how much we stay close to the book content in the assignments. Some of the Exercises that do not have a solution could be included there, overlap between book content and assignments reduces total study load. 
- The chapter is a bit low on visuals and interactive elements. The exercises related to derivations are useful but some illustrations or example implementations could be helpful to make the chapter more *real*. It could be kept for the off-season to be implemented for next year, in other aspects this chapter is already ahead of other chapters in the book, but there is potential for improvement on this front. See e.g. the interactive plots [here](https://mude.citg.tudelft.nl/book/2025/numerical_methods/6-initial-value-problem-singlestep.html) for inspiration. I have an old python FDM implementation of the 1D diffusion equation that I can dig up for you and propose in a separate branch after we merge this one.

mzijlema committed:
Update intro_to_pdes.md

small edit: hydraulic engineering -> civil engineering

mzijlema committed:
Update diffusion_equation.md

small edit: indicate that initial condition is positive

mzijlema committed:
Update advection_equation.md

small edit: remove statement "hydraulic engineering"

mzijlema committed:
Update diffusion_equation.md

edit: MOL -> method of lines

mzijlema committed:
Update advection_equation.md

edit: MOL -> method of lines

mzijlema commented:
I agree with the first five comments. I wanted to edit them, but apparently they weren't taken into account. Would it be possible for you to edit them? (Concerning the fourth comment: the statement w.r.t. "hydraulic engineering" can be removed.)

I also agree to include a bit more interactive elements. I will try it to include some simple python implementations.

I think that Section 1.4 is valuable because of the role of diffusion (both physical and numerical or either of them) with respect to diminishing wiggles and boundary layers, which come back later in the B modules/elective courses/cross-overs. This is something that cannot simply be explained in Sections 1.2 and 1.3. My idea is to discuss these aspects during the assignments (WA/GA) but not in the exam. I think this might be helpful to reduce the study load.

fmeer committed:
Use mathbf for vectors and matrices

fmeer committed:
Merge branch '2025-draft' into zijlema-review-q2-w1

fmeer committed:
Remove explicit reference to civil

fmeer committed:
Some missing mathbf's

mzijlema committed:
Update advection_diffusion_eq.md

small edits: MOL -> method of lines; explicit -> forward and add text at the end of section

mzijlema committed:
Update advection_equation.md

small edit: explicit Euler -> forward Euler

mzijlema committed:
Update advection_equation.md

add solution to the question concerning the accuracy of the leapfrog scheme

mzijlema committed:
Update advection_equation.md

small edit

mzijlema committed:
Update intro_to_pdes.md

edit: add advection-diffusion equation and explanation

fmeer reviewed (approved):
As discussed, I will merge this one now. The chapter is in a good state already. Possible improvements can go into a new branch.

fmeer committed:
Merge branch '2025-draft' into zijlema-review-q2-w1
